src/overlays/actors/ovl_Bg_Kin2_Bombwall/z_bg_kin2_bombwall.c

```diff
@@ -28,50 +40,181 @@ const ActorInit Bg_Kin2_Bombwall_InitVars = {
     (ActorFunc)BgKin2Bombwall_Draw,
 };
 
-// static ColliderCylinderInit sCylinderInit = {
-static ColliderCylinderInit D_80B6E6F0 = {
+ColliderCylinderInit D_80B6E6F0 = {
     { COLTYPE_NONE, AT_NONE, AC_ON | AC_TYPE_PLAYER, OC1_NONE, OC2_NONE, COLSHAPE_CYLINDER, },
     { ELEMTYPE_UNK0, { 0x00000000, 0x00, 0x00 }, { 0x00000008, 0x00, 0x00 }, TOUCH_NONE | TOUCH_SFX_NORMAL, BUMP_ON, OCELEM_NONE, },
     { 60, 60, 0, { 0, 0, 0 } },
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80B6E748[] = {
+Color_RGBA8 D_80B6E71C = { 0xD2, 0xD2, 0xD2, 0xFF };
+Color_RGBA8 D_80B6E720 = { 0x8C, 0x8C, 0x8C, 0xFF };
+
+Vec3f D_80B6E724 = { 0.0f, 0.33f, 0.0f };
+
+s8 D_80B6E730[] = { -0x3C, -0x22, -8, 0x12, 0x2C};
+s16 D_80B6E738[] = { 0x19, 0x17, 0x15, 0x13, 0x11, 0xF, 0xD, 0xA };
+
+InitChainEntry D_80B6E748[] = {
     ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneScale, 200, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneDownward, 300, ICHAIN_CONTINUE),
     ICHAIN_VEC3F_DIV1000(scale, 1000, ICHAIN_STOP),
-};
-
-#endif
-
-extern ColliderCylinderInit D_80B6E6F0;
-extern InitChainEntry D_80B6E748[];
-
-extern UNK_TYPE D_06000128;
-extern UNK_TYPE D_06000360;
-extern UNK_TYPE D_06000490;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Kin2_Bombwall/func_80B6E020.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Kin2_Bombwall/func_80B6E090.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Kin2_Bombwall/BgKin2Bombwall_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Kin2_Bombwall/BgKin2Bombwall_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Kin2_Bombwall/func_80B6E4B8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Kin2_Bombwall/func_80B6E4CC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Kin2_Bombwall/func_80B6E544.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Kin2_Bombwall/func_80B6E558.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Kin2_Bombwall/func_80B6E5F8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Kin2_Bombwall/func_80B6E614.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Kin2_Bombwall/BgKin2Bombwall_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Kin2_Bombwall/BgKin2Bombwall_Draw.s")
+};  
+
+//suggested name: BgKin2Bombwall_HadNearbyACCollision
+//The distance here actually matters: 
+//replacing 6400.0f by a distance too large makes the wall explode from a bomb quite far away.
+s32 func_80B6E020(BgKin2Bombwall *arg0, GlobalContext *arg1) {
+    Actor *bombwallCollider;
+
+    if ((arg0->collider.base.acFlags & 2) != 0) {
+        bombwallCollider = arg0->collider.base.ac;
+        if ((bombwallCollider != 0) && 
+            (Math3D_Vec3fDistSq(&arg0->dyna.actor.world.pos, &bombwallCollider->world.pos) < 6400.0f)) {
+            return 1;
+        }
+    }
+    return 0;
+}
```

Comment:
```suggestion
s32 func_80B6E020(BgKin2Bombwall *this, GlobalContext *globalCtx) {
    Actor *bombwallCollider;

    if (this->collider.base.acFlags & 2) {
        bombwallCollider = this->collider.base.ac;
        if ((bombwallCollider != NULL) && 
            (Math3D_Vec3fDistSq(&this->dyna.actor.world.pos, &bombwallCollider->world.pos) < 6400.0f)) {
            return true;
        }
    }
    return false;
}
```
The specific actor is usually called `this` and you can rename it in the params, same with globalCtx, makes it easier to read later so we dont need to look up what `arg0` is
I know its returning a s32, but the devs used true/false in different integer types, this is clearly a true/false return from my experience.
A conditional expression like `(value & 2)` will get evaluated down to `true or false` without the explicit !=0, so in this case we can just remove it.